outline-summaryst:0.1.0

aarneng opened the pull request:
<!--
Thanks for submitting a package! Please read and follow the submission guidelines detailed in the repository's README and check the boxes below. Please name your PR as `name:version` of the submitted package.

If you want to make a PR for something other than a package submission, just delete all this and make a plain PR.
-->

I am submitting
- [X] a new package
- [ ] an update for a package

<!--
Please add a brief description of your package below and explain why you think it is useful to others. If this is an update, please briefly say what changed.
-->

Description: `outline-summaryst` is a basic template designed for including a summary for each entry in the table of contents, particularly useful for writing books. It provides a simple structure for organizing content and generating formatted documents with summary sections.

<!--
These things need to be checked for a new submission to be merged. If you're just submitting an update, you can delete the following section.
-->

I have read and followed the submission guidelines and, in particular, I
- [X] selected a name that isn't the most obvious or canonical name for what the package does
- [X] added a `typst.toml` file with all required keys
- [X] added a `README.md` with documentation for my package
- [X] have chosen a license and added a `LICENSE` file or linked one in my `README.md`
- [X] tested my package locally on my system and it worked
- [X] `exclude`d PDFs or README images, if any, but not the LICENSE

<!--
The following box only needs to be checked for **template** submissions. If you're submitting a package that isn't a template, you can delete the following section. See the guidelines section about licenses in the README for more details.
-->
- [x] ensured that my package is licensed such that users can use and distribute the contents of its template directory without restriction, after modifying them through normal use.

laurmaedje commented:
From the description, it seems to me that this is more a building block rather than a complete template. Would it perhaps make more sense for it be just a package, not a template package?

aarneng commented:
> From the description, it seems to me that this is more a building block rather than a complete template. Would it perhaps make more sense for it be just a package, not a template package?

I'm honestly not sure, I guess it's currently kind of in the middle of the two. The main idea is that it implements a way to add a summary into the `#outline()`,  but also changes the styling of said outline and each heading. Here is a link to a version running on the web app, if you want to quickly look through it yourself: https://typst.app/project/rREF3kavhUHhGJqNFgHETM

If you feel that the current state makes it more of a package than a template, I can make changes re-submit it as that. If so, do you have any pointers on how to make it more template-like for future versions?

laurmaedje commented:
Conceptually, it does feel more like a package to me. Of course, right now some of it is set up like a template, which would need to be changed if it were just a package. 

> If so, do you have any pointers on how to make it more template-like for future versions?

Is that desirable? Being a template mostly means that it's not mix-and-match with other templates. If it's a package that can be flexibly used alongside other stuff, great!

But ultimately, it's totally up to you whether you want to go with this, those are just my 2ct.

aarneng commented:
I think that's fair, i've changed this and removed the template part. I hadn't considered that making it a template limits usability to a degree, that is nice feedback :)

laurmaedje commented on packages/preview/outline-summaryst/0.1.0/src/outline-summaryst.typ:
```diff
@@ -0,0 +1,123 @@
+#let supl = [Custom Heading]
+
+
+#let outline-summaryst(
```
Is this function needed at all? Wouldn't it make more sense to provide a function that generates just the fancy outline instead? 

Since your headings use a special `make-heading` function instead of the syntax, the heading show rule could also be included with that. That way it would be even more composable.

aarneng replied:
I've changed the behavior of the both functions to be more composable based on your feedback (note that I renamed `outline-summaryst` to `style-outline`).

To answer your question of why `outline-summaryst` is not a function that generates the fancy outline, this is because some templates include an `#outline()` by default, so this package could not be used alongside such templates. The current behavior, where we have a show rule for `outline.entry`, solves this issue

laurmaedje commented:
Thanks!
